mobile/library/common/jni/jni_utility.cc

```diff
@@ -135,19 +130,17 @@ jlongArray native_final_stream_intel_to_array(JniHelper& jni_helper,
 }
 
 jobject native_map_to_map(JniHelper& jni_helper, envoy_map map) {
-  jclass jcls_hashMap = jni_helper.getEnv()->FindClass("java/util/HashMap");
-  jmethodID jmid_hashMapInit = jni_helper.getMethodId(jcls_hashMap, "<init>", "(I)V");
-  jobject j_hashMap = jni_helper.getEnv()->NewObject(jcls_hashMap, jmid_hashMapInit, map.length);
+  LocalRefUniquePtr<jclass> jcls_hashMap = jni_helper.findClass("java/util/HashMap");
+  jmethodID jmid_hashMapInit = jni_helper.getMethodId(jcls_hashMap.get(), "<init>", "(I)V");
+  jobject j_hashMap =
+      jni_helper.getEnv()->NewObject(jcls_hashMap.get(), jmid_hashMapInit, map.length);
   jmethodID jmid_hashMapPut = jni_helper.getMethodId(
-      jcls_hashMap, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+      jcls_hashMap.get(), "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
```

Comment:
nit: not super important and not for this PR, but perhaps at some point we may want to consider adding the dereferencing operator (i.e. `*`) to LocalRefUniquePtr so we can use it instead of `.get()`.  OTOH, `.get()` makes it clearer, so I'm indifferent either way, just thought I'd mention as food for thought.

Reply:
`std::unique_ptr` already implements `operator *()`, but it will dereference it whereas [std::unique_ptr<>::get()](https://en.cppreference.com/w/cpp/memory/unique_ptr/get) will simply return the borrowed pointer.

Reply:
I'm slightly confused by this. It looks like the method signature for `getMethodId()` is:
```
jmethodID getMethodId(jclass clazz, const char* name, const char* signature);
```
The type of `jcls_hashMap` is `LocalRefUniquePtr<jclass>` so I would expect we would want to pass `*jcls_hashMap` into `getMethodId` since we want a `jclass` not `jclass*`. But clearly your code works so I must be misunderstanding. Can you explains?

Reply:
The JNI types are extremely confusing.

Here's the definition.
```
typedef _jclass *jclass;
```

```
LocalRefUniquePtr<jclass> foo = jni_helper.findClass("...");
jclass jc1 = foo.get(); // will return jclass
_jclass jc2 = *foo; // deference, will return _jclass instead of jclass
```

Reply:
Oh, ffs! That's crazy. Thanks, java :)